src/derivable/mixins/take.ts

```diff
@@ -0,0 +1,86 @@
+import { Derivable, MaybeFinalState, TakeOptions, TakeOptionValue } from '../../interfaces';
+import { unresolved } from '../../symbols';
+import { equals, ErrorWrapper, FinalWrapper } from '../../utils';
+import { Atom } from '../atom';
+import { BaseDerivable } from '../base-derivable';
+import { Derivation } from '../derivation';
+import { unwrap } from '../unwrap';
+
+const true$ = new Atom(FinalWrapper.wrap(true));
+const false$ = new Atom(FinalWrapper.wrap(false));
+const skipped = Symbol('skipped');
+const finalSkipped = FinalWrapper.wrap(skipped);
+
+export function takeMethod<V>(this: Derivable<V>, { from, when, until, once, stopOnError, skipFirst }: Partial<TakeOptions<V>>): Derivable<V> {
+    // From is true by default, once it becomes true, it will stay true.
+    const from$ = toMaybeDerivable(from, this, true, true);
+    // When is true by default.
+    const when$ = toMaybeDerivable(when, this, true);
+    // Until is false by default, once it becomes true, it will stay true.
+    const until$ = toMaybeDerivable(until, this, false, true);
+    // SkipFirst is false by default, once it becomes false again, it will stay false.
+    const skipFirstState$ = skipFirst ? new Atom<MaybeFinalState<V> | typeof skipped>(unresolved) : undefined;
+
+    if (!from$ && !when$ && !until$ && !once && !stopOnError && !skipFirstState$) {
+        return this;
+    }
+
+    const previousState$ = (until$ || when$) && new Atom<V>(unresolved);
+    function resultingState(value: MaybeFinalState<V>) {
+        previousState$ && previousState$.set(value);
+        return value;
+    }
+
+    return new Derivation<V>(() => {
+        if (from$ && !from$.get()) {
+            return resultingState(unresolved);
```

Comment:
Is it needed to go through `resultingState()` here? Since the `Atom` that may exist already is `unresolved`.

Reply:
You are right, because `previousState$` is an Atom and it will be reset on rollback in a transaction it has no effect. The other `resultingState`s are not tested properly though, I don't see a failing test when I remove those. Looking into it now.